connectors/source.py

```diff
@@ -693,6 +693,11 @@ def get_file_extension(self, filename):
         return get_file_extension(filename)
 
     def can_file_be_downloaded(self, file_extension, filename, file_size):
+        return self.is_valid_file_type(
+            file_extension, filename
+        ) and self.is_file_size_within_limit(file_size, filename)
```

Comment:
I thought one of the point for the extraction service is to remove this file size limitation?

Reply:
In the func `is_file_size_within_limit` we are checking if the user has `use_text_extraction_service` enabled or not. If they have it enabled, we don't care about file size.

This pre-check is a generic gateway before any file downloading.